src/sidebar/components/user-menu.js

```diff
@@ -11,15 +11,32 @@ import MenuItem from './menu-item';
 import MenuSection from './menu-section';
 import SvgIcon from '../../shared/components/svg-icon';
 
+/** @typedef {import('../../types/config').SidebarConfig} SidebarConfig */
+
+/** @typedef {import('../../types/config').HostConfig} HostConfig */
+
+/** @typedef {import('../../types/auth').AuthState} AuthState */
+
+/**
+ * @typedef UserMenuProps
+ * @prop {AuthState} auth - object representing authenticated user and auth status
+ * @prop {() => any} onLogout - onClick callback for the "log out" button
+ * @prop {Object} bridge - services
+ * @prop {(name: string, params?: Object<'tag', string>) => any} serviceUrl - Services
+ * @prop {SidebarConfig} settings
+ */
+
 /**
  * A menu with user and account links.
  *
  * This menu will contain different items depending on service configuration,
  * context and whether the user is first- or third-party.
+ *
+ * @param {UserMenuProps} props
  */
 function UserMenu({ auth, bridge, onLogout, serviceUrl, settings }) {
   const isThirdParty = isThirdPartyUser(auth.userid, settings.authDomain);
-  const service = serviceConfig(settings);
+  const service = serviceConfig(/** @type {HostConfig} */ (settings));
```

Comment:
I'm forced to cast here, but I think this is a mistake. I'm pretty sure serviceConfig() only cares about a HostConfig object upon inspecting it, but this is clearly a SidebarConfig object.

Reply:
Have a read through the comments in `types/config.js`. The `settings` object's type is `MergedConfig`, which is a combination of `SidebarConfig` and `HostConfig`, rather than `SidebarConfig`.

Reply:
thats exactly what I was looking for, just forgot about it!